app/src/ui/dispatcher/error-handlers.ts

```diff
@@ -690,6 +690,59 @@ export async function schannelUnableToCheckRevocationForCertificate(
   return null
 }
 
+/**
+ * Handler for when an action the user attempts cannot be done because there are local
+ * changes that would get overwritten.
+ */
+export async function localChangesOverwrittenHandler(
+  error: Error,
+  dispatcher: Dispatcher
+): Promise<Error | null> {
+  const e = asErrorWithMetadata(error)
+  if (e === null) {
+    return error
+  }
+
+  const gitError = asGitError(e.underlyingError)
+  if (gitError === null) {
+    return error
+  }
+
+  const dugiteError = gitError.result.gitError
+  if (dugiteError === null) {
+    return error
+  }
+
+  if (
+    dugiteError !== DugiteError.LocalChangesOverwritten &&
```

Comment:
It's not immediately obvious tot me why we'd be including the `LocalChangesOverwritter` error in this handler. Wouldn't that get caught by the `localChangesOverwrittenOnCheckoutHandler`?

Reply:
`localChangesOverwrittenOnCheckoutHandler()` ignores errors that don't get triggered by a checkout, see [its comment](
 https://github.com/desktop/desktop/blob/development/app/src/ui/dispatcher/error-handlers.ts#L466-L475):

> This indicates to us whether the action which triggered the
> LocalChangesOverwritten was the AppStore _checkoutBranch method.
> Other actions that might trigger this error such as deleting
> a branch will not provide this specific gitContext and that's
> how we know we can safely move the changes to the destination
> branch.

There may not be many situations where this happens, but I thought it would still be nice to catch them (I'm going to add a comment in the code).